Here, I have imported some Libraries:

In [ ]:
import pandas as pd
import numpy as np


In [ ]:
from sklearn.datasets import load_iris
iris_dataset = load_iris()

In [ ]:
print("Keys of iris_dataset: \n{}".format(iris_dataset.keys()))

Keys of iris_dataset: 
dict_keys(['data', 'target', 'target_names', 'DESCR', 'feature_names', 'filename'])


The value of the key DESCR is a short description of the dataset. Lets look at what it contains: 

In [ ]:
print(iris_dataset['DESCR'][:193]+ "\n...")

.. _iris_dataset:

Iris plants dataset
--------------------

**Data Set Characteristics:**

    :Number of Instances: 150 (50 in each of three classes)
    :Number of Attributes: 4 numeric, pre
...


Ass we can see it contains some information about the dataset itself.

In [ ]:
print("Target names: {}".format(iris_dataset['target_names']))

Target names: ['setosa' 'versicolor' 'virginica']


Next in line is the feature_names 
The value of feature_names is a list of strings, giving the description of each feature it includes the 'sepal length','sepal width', 'petal length' and 'petal width' all in centimeters. 

In [ ]:
print("Feature names: \n{}".format(iris_dataset['feature_names']))

Feature names: 
['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


The data that we are going to use is itself contained in the target and data foelds. The data contains numeric measurements of sepal length, sepal width,petal length and petal width in a numpy array:

In [ ]:
print("Type of data: {}".format(type(iris_dataset['data'])))

Type of data: <class 'numpy.ndarray'>


The rows in this data array correspond to the flowers, while the columns represent the four measurements that were taken for each flower:

In [ ]:
print("Shape of data: {}".format(iris_dataset['data'].shape))

Shape of data: (150, 4)


We see that the array contains measurement for 150 different flowers.The indivual items are called samples/data points in machine learning and their properities are called features.So we have 150 data points and 4 features. The shape of the data array is the number of samples multiplied by the number of features

In [ ]:
print("First five columns of data:\n{}".format(iris_dataset['data'][:5]))

First five columns of data:
[[5.1 3.5 1.4 0.2]
 [4.9 3.  1.4 0.2]
 [4.7 3.2 1.3 0.2]
 [4.6 3.1 1.5 0.2]
 [5.  3.6 1.4 0.2]]


From this data,we can see that all of the first five flowers have a petal width of 0.2cm and that the first flower has the longest sepal, at 5.1 cm. We will be making this output in the form of a table.

In [ ]:
print("Type of target: {}".format(type(iris_dataset['target'])))

Type of target: <class 'numpy.ndarray'>


Target ia a one dimensional array on seeing the shape we can see that it conatins one entry per flower. 

In [ ]:
print("Shape of Target: {}".format(iris_dataset['target'].shape))

Shape of Target: (150,)


On seeing the target key and exploring the values we can see that The species are encoded as integers from 0 to 2

In [ ]:
print("Target:\n{}".format(iris_dataset['target']))

Target:
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 2
 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2
 2 2]


The meanings of the numbers are given by the iris['target_names'] array:0 means setosa, 1 means versicolor and 2 means virginica

Training and Testing data

In [ ]:
from sklearn.model_selection import train_test_split 
X_train, X_test, y_train, y_test = train_test_split(
    iris_dataset['data'],iris_dataset['target'],random_state=0
)

The ouput of the train_test_split function is X_train,X_test,y_train,y_test which are all Numpy arrays.X_train contains 75% of the rows of thhe dataset and X_test contains the remaining 25%.

In [ ]:
print("X_train shape: {}".format(X_train.shape))
print("y_train shape: {}".format(y_train.shape))
print("X_test shape: {}".format(X_train.shape))
print("y_test shape: {}".format(y_test.shape))

X_train shape: (112, 4)
y_train shape: (112,)
X_test shape: (112, 4)
y_test shape: (38,)


Inspecting the data:

In [ ]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors=1)

This is the KNN machine learning model. The most important parameter of Kneighbor sClassifier is the number of neighbors which we will set to 1 

In [ ]:
knn.fit(X_train,y_train)

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
                     metric_params=None, n_jobs=None, n_neighbors=1, p=2,
                     weights='uniform')

Making Predictions

In [ ]:
X_new = np.array([[5,2.9,1,0.2]])
print("X_new.shape: {}".format(X_new.shape))

X_new.shape: (1, 4)


In [ ]:
prediction = knn.predict(X_new)

print("Prediction: {}".format(prediction))
print("Predicted target name: {}".format(
    iris_dataset['target_names'][prediction]
))


Prediction: [0]
Predicted target name: ['setosa']


Evaluating the model:

In [ ]:
y_pred = knn.predict(X_test)
print("Test set predictions:\n {}".format(y_pred))

Test set predictions:
 [2 1 0 2 0 2 0 1 1 1 2 1 1 1 1 0 1 1 0 0 2 1 0 0 2 0 0 1 1 0 2 1 0 2 2 1 0
 2]


In [ ]:
print("Test set score: {}".format(np.mean(y_pred == y_test)))

Test set score: 0.9736842105263158


Here the accuracy of the Machine Learning  model is 97.36%